pkl-core/src/main/java/org/pkl/core/util/IoUtils.java

```diff
@@ -663,6 +663,69 @@ public static URI fixTripleSlashUri(URI baseUri, URI newUri) {
     return newUri;
   }
 
+  /**
+   * Windows reserves characters {@code <>:"\|?*} in filenames.
+   *
+   * <p>For any such characters, enclose their decimal character code with parentheses. Verbatim
+   * {@code (} is encoded as {@code ((}.
+   */
+  public static String encodePath(String path) {
+    if (path.isEmpty()) return path;
+    var sb = new StringBuilder();
+    for (var i = 0; i < path.length(); i++) {
+      var character = path.charAt(i);
+      switch (character) {
+        case '<', '>', ':', '"', '\\', '|', '?', '*' -> {
+          sb.append('(');
+          sb.append(ByteArrayUtils.toHex(new byte[] {(byte) character}));
+          sb.append(")");
+        }
+        case '(' -> sb.append("((");
+        default -> sb.append(path.charAt(i));
+      }
+    }
+    return sb.toString();
+  }
+
+  /** The inverse of {@link #encodePath(String)} */
```

Comment:
Technically, this function is not really the inverse of `encodePath`, as it decodes anything, not only the chars that `encodePath` encodes. So one can decode `(66)(6f)(6f).text` as `foo.txt` which cannot be generated by `encodePath`. But I'm being pedantic and I think it's fine that it decodes anything.

Reply:
Good point. Although, I'm actually getting rid of this altogether; this method isn't used anywhere.